pkg/registry/core/node/strategy.go

```diff
@@ -267,13 +267,15 @@ func ResourceLocation(getter ResourceGetter, connection client.ConnectionInfoGet
 	return &url.URL{Scheme: schemeReq, Host: net.JoinHostPort(info.Hostname, portReq)}, proxyTransport, nil
 }
 
-func dynamicKubeletConfigIsDeprecatedWarning(obj runtime.Object) []string {
+func fieldIsDeprecatedWarnings(obj runtime.Object) []string {
 	newNode := obj.(*api.Node)
+	var warnings []string
 	if newNode.Spec.ConfigSource != nil {
-		var warnings []string
 		// KEP https://github.com/kubernetes/enhancements/issues/281
 		warnings = append(warnings, "spec.configSource: the feature is removed")
-		return warnings
 	}
-	return nil
+	if len(newNode.Spec.DoNotUseExternalID) > 0 {
+		warnings = append(warnings, "spec.externalID: this field is deprecated, and will be removed in a future release")
```

Comment:
the field will *not* be removed, but it is also unused by kubernetes

Reply:
Thanks @liggitt , the warning message has been updated.